concepts/0519-goal-codes/README.md

```diff
@@ -0,0 +1,149 @@
+* `cci.healthcare.arrange`
+
+Goals are inherently self-attested. Thus, goal codes don't represent objective fact that a recipient can rely upon in a strong sense; subsequent interactions can always yield surprises. Even so, goal codes let agents triage interactions and find misalignments early; there's no point in engaging if their goals are incompatible. This has significant benefits for spam prevention, among other things.
+
+### Verbs
+
+Notice the verbs in the examples: `sell`, `date`, `hire`, and `arrange`. Goals typically involve action; a complete goal code should have one or more verbs in it somewhere. Turning verbs into nouns (e.g., `employment.references` instead of `employment.check-references`) is considered bad form. (Some namespaces may put the verbs at the end; some may put them in the middle. That's a purely stylistic choice.)
+
+### Directionality
+
+Notice, too, that the verbs may imply directionality. A goal with the `sell` verb implies that the person announcing the goal is a would-be seller, not a buyer. We could imagine a more general verb like `engage-in-commerce` that would allow either behavior. However, that would often be a mistake. The value of goal codes is that they let agents align around intent; announcing that you want to engage in general commerce without clarifying whether you intend to sell or buy may be too vague to help the other party make decisions. 
+
+It is conceivable that this would lead to parallel branchs of a goal ontology that differ only in the direction of their verb. Thus, we could imagine `sell.A` and `sell.B` being shadowed by `buy.A` and `buy.B`. This might be necessary if a family of protocols allow either party to initiate an interaction and declare the goal, and if both parties view the goals as perfect mirror images. However, practical considerations may make this kind of parallelism unlikely. A random party contacting an individual to sell something may need to be quite clear about the type of selling they intend, to make it past a spam filter. In contrast, a random individual arriving at the digital storefront of a mega retailer may be quite vague about the type of buying they intend. Thus, the `buy.*` side of the namespace may need much less detail than the `sell.*` side.
+
+### Goals for others
+
+Related to directionality, it may occasionally be desirable to propose goals to others, rather than adovcating your own: "Let &lt;parties = us = Alice, Bob, and Carol> &lt;goal = hold an auction&gt; -- I nominate Carol to be the &lt;role = auctioneer&gt; and get us started." The difference between a normal message and an unusual one like this is not visible in the goal code; it should be exposed in additional fields that associate the goal with a particular DID+role pair. Essentially, you are proposing a goal to another party, and these extra fields clarify who should receive the proposal, and what role/perspective they might take with respect to the goal.
```

Comment:
I would generalize DID to "ID" in "DID+role pair". Not all identifiers in DIDComm are DIDs.

Reply:
Updated to be more general.